### DISCUSSION TITLE
How to specify APP_API_TOKEN dapr env variable

### DISCUSSION THREAD

**Discussion by mzj** [microsoft/aspire#5067] [Q&A]
Hi,

I'm trying to set the APP_API_TOKEN Dapr environment variable to a specific value, but I couldn't find a way to do that using the x.WithDaprSidecar("app-id", new DaprSidecarOptions { }) builder method. I suspect I might be missing something obvious.

By the way, I'm using Aspire locally only, if that matters.

**Comment by JefSchraag** [comment]
If the `IDaprSidecarResource` would implement `IResourceWithEnvironment` then you could have used:

```c#
.WithDaprSidecar(daprBuilder =>
{
    daprBuilder.WithOptions(daprOptions);
    daprBuilder.WithEnvironment("APP_API_TOKEN", daprApiToken);
}
```
But with the current version (9.7.0) of the CommunityToolkit.Aspire.Hosting.Dapr, the above code does not compile.

As a workaround, you can set the environment variable via an `EnvironmentCallbackAnnotation`.

I am using this extension method:

```c#
public static class DaprExtensions
{
	/// <summary>
	/// Adds a Dapr sidecar with the specified options and an APP_API_TOKEN environment variable
	/// </summary>
	public static IResourceBuilder<ProjectResource> WithDaprSidecarAndToken(
		this IResourceBuilder<ProjectResource> builder,
		DaprSidecarOptions daprOptions,
		string daprApiToken = "dev-secret-token")
	{
		return builder
			.WithEnvironment("APP_API_TOKEN", daprApiToken)
			.WithDaprSidecar(daprBuilder =>
			{
				daprBuilder.WithOptions(daprOptions);
				// Direct annotation approach - works with any IResource
				daprBuilder.Resource.Annotations.Add(new EnvironmentCallbackAnnotation((envVars) =>
				{
					envVars["APP_API_TOKEN"] = daprApiToken;
				}));
			});
	}
}
```

**Comment by davidfowl** [comment]
The dapr resource has moved to the community toolikit https://github.com/CommunityToolkit/Aspire/issues
